pkg/route/engine.go

```diff
@@ -901,6 +901,14 @@ func (engine *Engine) HasServer(name string) bool {
 	return engine.protocolSuite.Get(name) != nil
 }
 
+// SetBasePath To modify basePath
+func (engine *Engine) SetBasePath(basePath string) {
```

Comment:
you should follow  `pull request template` instead of a sentence

Reply:
Can you describe the scenario and reason for adding this method?

Reply:
> Can you describe the scenario and reason for adding this method? @springrain
Similar to the java web's contextPath, it is convenient for nginx distribution

Reply:
I think using `route.Group`  will work instead of it

Reply:
> I think using `route.Group` will work instead of it

Group strongly relies on development specifications, and it is not necessarily easy to modify, when multi-project integration, context-path is very useful, I currently use reflection to solve it temporarily.
```go
//需要在路由初始化前调用
func SetContextPath(contextPath string) {
	if contextPath == "" {
		return
	}
	//获取引擎
	h := WebEngine()
	//因为Engine匿名注入了RouterGroup,所以直接获取Engine的反射对象
	engine := reflect.ValueOf(h).Elem()
	//获取RouterGroup的basePath属性反射值对象
	basePath := engine.FieldByName("basePath")
	//获取basePath的UnsafeAddr
	p := unsafe.Pointer(basePath.UnsafeAddr())
	//重新赋值basePath的反射值,NewAt默认返回的是指针,使用Elem获取反射值对象
	basePath = reflect.NewAt(basePath.Type(), p).Elem()
	//设置反射值
	basePath.Set(reflect.ValueOf(contextPath))
}
```

Reply:
OK and need to add unit test.